### ISSUE TITLE
ERROR: Failed to download http://www.cpan.org/src/5.0/perl-5.16.3.tar.bz2

### ISSUE THREAD

**Issue by gideondsouza** [gugod/App-perlbrew#361]
So I get the following when I try to install a particular version with perlbrew:

`ERROR: Failed to download http://www.cpan.org/src/5.0/perl-5.16.3.tar.bz2`

I checked the URL in my browser and it gives me a 503. All available versions of perl are giving me 503. 
- Is there something I'm missing?
- Is the source site down temporarily?

**Comment by mstreuhofer** [comment]
well if it was then it was an issue with www.cpan.org, not perlbrew.
please close this issue.

**Comment by msoutopico** [comment]
In my case the issue happens when I run command

    » perlbrew --force install perl-5.16.0
    Fetching perl 5.16.0 as /opt/perlbrew/dists/perl-5.16.0.tar.bz2
    Download http://www.cpan.org/src/5.0/perl-5.16.0.tar.bz2 to /opt/perlbrew/dists/perl-5.16.0.tar.bz2
    ERROR: Failed to download http://www.cpan.org/src/5.0/perl-5.16.0.tar.bz2

However, I can download file http://www.cpan.org/src/5.0/perl-5.16.0.tar.bz2 through the browser with no issues. 

In other words, the file seems available for download in CPAN but perlbrew cannot install it. It doens't seem like the issue is with CPAN.

**Comment by msoutopico** [comment]
Some more output:

    » perlbrew --force install perl-5.16.0
    Installing /opt/perlbrew/build/perl-5.16.0 into /opt/perlbrew/perls/perl-5.16.0
    
    This could take a while. You can run the following command on another shell to track the status:
    
      tail -f /opt/perlbrew/build.perl-5.16.0.log    
    
    Installation process failed. To spot any issues, check
    
      /opt/perlbrew/build.perl-5.16.0.log
    
    If some perl tests failed and you still want install this distribution anyway,
    do:
    
      (cd /opt/perlbrew/build/perl-5.16.0; make install)
    
    You might also want to try upgrading patchperl before trying again:
    
      perlbrew install-patchperl
    
    Generally, if you need to install a perl distribution known to have minor test
    failures, do one of these command to avoid seeing this message
    
      perlbrew --notest install perl-5.16.0
      perlbrew --force install perl-5.16.0
    
    » cat /opt/perlbrew/build.perl-5.16.0.log
    Auto-guessed '5.16.0'
    No patch utility found
    patching ext/Errno/Errno_pm.PL
    » perlbrew install-patchperl
    
    /opt/perlbrew/bin/patchperl already exists, are you sure to override ? [y/N] [N] Y
    
    patchperl is installed to
    
        /opt/perlbrew/bin/patchperl
    
    
Conclusion: I can't use perl. 

Any suggestions?  Thanks in advance.

**Comment by gugod** [comment]
@msoutopico Was it about downloading failure, or building failures ? You could try removing tarballs under `~/perl5/perlbrew/dists/` and retry the "download" command. I guess in you case it'd be `/opt/perlbrew/dists` .

OTOH, you could also try downloading the tarball with browser, then invoke the installation command with the path to tarball: `perlbrew install ~/Downloads/perl-5.16.3.tar.gz` 

Those should help you isloating whether the issue is about downloading or bulding (if not both).

**Comment by seminull** [comment]
I see this happen if `perlbrew init` isn't run.  If this is the issue, perhaps a more helpful message is needed when `perlbrew install` fails for this reason.

**Comment by abelcallejo** [comment]
after `perlbrew install` one must do `perlbrew init` to make perlbrew work

**Comment by fbicknel** [comment]
I was able to download _perl-5.29.0.tar_ using `wget` (or my browser), but `$ perlbrew install perl-5.29.0` failed as mentioned above.

@gugod's comment above was helpful, but one can also move the tarball to `~/perl5/perlbrew/dists` and `perlbrew install perl-5.29.0` will find it there and not re-try the download. It will also compress the tarball in place, if not already compressed.

**Comment by j1n3l0** [comment]
I ran into this problem but it was because I did not have write access to the directory artefacts were being downloaded into.

**Comment by cwilson21** [comment]
@gideondsouza this also happens if you don't have `patch` installed. I was having the same issue and found that `patch` was missing
